dashboard/src/main/resources/templates/macros.ftl

```diff
@@ -0,0 +1,81 @@
+<#macro formatJarLinkageReport jarLinkageReport jarToDependencyPaths>
+  <h3>${jarLinkageReport.getJarPath().getFileName()?html}</h3>
+
+  <#assign causeToSourceClasses = jarLinkageReport.getCauseToSourceClasses() />
+  <#assign errorPlural = jarLinkageReport.getTotalErrorCount() gt 1 />
+  <#assign causePlural = causeToSourceClasses.keySet()?size gt 1 />
+  <#assign linkageErrors = jarLinkageReport.getTotalErrorCount()
+  + " linkage error" + errorPlural?string('s', '') />
+  <#assign classes = causeToSourceClasses.keySet()?size
+  + " class" + causePlural?string('es', '') />
+  <p class="jar-linkage-report">${linkageErrors} in ${classes}</p>
+
+  <#list causeToSourceClasses.keySet() as key >
+    <p class="jar-linkage-report-cause">${key?html} Referenced from</p>
+    <ul class="jar-linkage-report-cause">
+      <#list causeToSourceClasses.get(key) as sourceClass>
+        <li class="jar-linkage-report-source-class"><code>${sourceClass?html}</code></li>
```

Comment:
There doesn't appear to be a `jar-linkage-report-source-class` class. Could you use the selector `ul.jar-linkage-report-cause > li`?

Also, if that style already specifies monospaced font etc., do you still need the `<code>` tag?